service/history/shard/context_impl.go

```diff
@@ -1436,8 +1437,7 @@ func (s *ContextImpl) handleReadError(err error) error {
 	case *persistence.ShardOwnershipLostError:
 		// Shard is stolen, trigger shutdown of history engine.
 		// Handling of max read level doesn't matter here.
-		s.transition(contextRequestStop{})
-		return err
+		return multierr.Combine(err, s.transition(contextRequestStop{}))
```

Comment:
I don't think we can/should combine the errors here (and two more places below).
- Many places in our code path is still checking error type directly not via errors.As, so those places might break.
- To me, the error from s.transition is internal to the shard context impl and upper layer should not know. 

cc @dnr Would you mind also take a look?

Reply:
Yeah, this PR doesn't make much sense. I think you should revert these changes.

The error returned from transition is really only meaningful for contextRequestAcquired, the others will always return nil and the result doesn't have to be checked. I know errcheck isn't smart enough to figure that out but we can just manually ignore them.

As Yichao said, transition already logs so callers should not.

And I agree the multierr stuff is not appropriate here.